cpp/log/signer.cc

```diff
@@ -59,7 +63,12 @@ std::string Signer::RawSign(const std::string& data) const {
   unsigned int sig_size = EVP_PKEY_size(pkey_.get());
   unsigned char* sig = new unsigned char[sig_size];
 
-  CHECK_EQ(1, EVP_SignFinal(&ctx, sig, &sig_size, pkey_.get()));
+  if (synchronize_signing_) {
+    std::lock_guard<std::mutex> lock(lock_);
+    CHECK_EQ(1, EVP_SignFinal(&ctx, sig, &sig_size, pkey_.get()));
```

Comment:
I disagree with this, if `synchronize_signing_` is `const` (as I suggested earlier), there is effectively no difference, and this will only add complexity (there will *still* be two flows to check or fix if a bug is discovered!).

I think there should be some good comment here explaining why this is done, how this is not necessary in theory but is a workaround for the HSM threading issue, etc...